src/redux/books/books.js

```diff
@@ -1,26 +1,94 @@
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
 
+const apiUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Gqq4TbGrZmbMnwiy8Tsp/books';
+const GET_BOOKS = 'bookstore/books/GET_BOOKS';
 const initialState = [];
 
-export const addbook = (payload) => ({
-  type: ADD_BOOK,
-  payload,
+const getBooksFromApi = async () => {
+  const response = await fetch(apiUrl, {
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  const books = await response.json();
+  const booksData = Object.keys(books).map((id) => ({
+    id,
+    title: books[id][0].title,
+    author: books[id][0].author,
+  }));
+  return booksData;
+};
+
+export const getbooks = () => (async (dispatch) => {
+  const booksData = await getBooksFromApi();
+  dispatch({
+    type: GET_BOOKS,
+    payload: booksData,
+  });
+});
+
+const addBooksToApi = async (payload) => {
+  const {
+    id, title, author, category,
+  } = payload;
+  const send = await fetch(apiUrl, {
+    method: 'POST',
+    body: JSON.stringify({
+      item_id: id,
+      title,
+      author,
+      category,
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  (await send.text());
+};
+
+export const addbook = (payload) => (async (dispatch) => {
+  const { id, title, author } = payload;
+  await addBooksToApi(payload);
+  dispatch({
+    type: ADD_BOOK,
+    book: { id, title, author },
+  });
 });
 
-export const removeBook = (payload) => ({
-  type: REMOVE_BOOK,
-  payload,
+const removeBookFromApi = async (id) => {
+  const remove = await fetch(`${apiUrl}/${id}`, {
+    method: 'DELETE',
+    body: JSON.stringify({
+      item_id: id,
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  (await remove.text());
+};
+
+export const removeBook = (id) => (async (dispatch) => {
+  await removeBookFromApi(id);
+  dispatch({
+    type: REMOVE_BOOK,
+    id,
+  });
 });
```

Comment:
- I suggest you apply the same advise above to remove books from the API. Please use just one method or function to perform both your API calls and your dispatch actions 🙏